rclcpp/include/rclcpp/experimental/serialized_message.hpp

```diff
@@ -0,0 +1,90 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__EXPERIMENTAL__SERIALIZED_MESSAGE_HPP_
+#define RCLCPP__EXPERIMENTAL__SERIALIZED_MESSAGE_HPP_
+
+#include <cstring>
+
+#include "rcutils/logging_macros.h"
+
+#include "rmw/serialized_message.h"
+
+namespace rclcpp
+{
+namespace experimental
+{
+
+/// Object oriented version of rcl_serialized_message_t with destructor to avoid memory leaks
+class SerializedMessage : public rcl_serialized_message_t
+{
+public:
+  SerializedMessage()
+  : rcl_serialized_message_t(rmw_get_zero_initialized_serialized_message())
+  {}
+
+  explicit SerializedMessage(const SerializedMessage & sc)
```

Comment:
can you simply call the other constructor (`SerializedMessage(const rcl_serialized_message_t & sc)`? Alternatively provide a generic `init` or `swap` function to de-duplicate the code within the two constructors.

Reply:
just removed it, as implicit conversion is possible